### PR TITLE
Use k8s.io/client-go v0.17.0

### LINKED ISSUE #34: Use k8s.io/client-go v0.17.0 instead of github.com/ericchiang/k8s

### PR DESCRIPTION
**Pull request by vitorarins**
Fixes #34 

Also some refactoring was made in main function.

Perhaps is easier to review via this commit:  https://github.com/estafette/estafette-letsencrypt-certificate/commit/19c50254953ed54cdf5e5dc6cabe80a51ddd4e44

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -303,27 +315,27 @@ func makeSecretChanges(kubeClient *k8s.Client, secret *corev1.Secret, initiator
...
 		}
 		letsEncryptUser.key = privateKey
 
-		log.Info().Msgf("[%v] Secret %v.%v - Creating lego config...", initiator, *secret.Metadata.Name, *secret.Metadata.Namespace)
+		log.Info().Msgf("[%v] Secret %v.%v - Creating lego config...", initiator, secret.Name, secret.Namespace)
 		config := lego.NewConfig(&letsEncryptUser)
 
-		// create letsencrypt lego client
-		log.Info().Msgf("[%v] Secret %v.%v - Creating lego client...", initiator, *secret.Metadata.Name, *secret.Metadata.Namespace)
+		// create letsencrypt lego clientset
```
Comment (JorritSalverda) [inline main.go:329]: This is an unintended replacement it seems

File: main.go [hunk 2]
```diff
@@ -367,15 +379,15 @@ func makeSecretChanges(kubeClient *k8s.Client, secret *corev1.Secret, initiator
...
 		// 	err = cloudflareProvider.CleanUp(hostname, "", "123d==")
 		// 	if err != nil {
-		// 		log.Info().Err(err).Msgf("[%v] Secret %v.%v - Cleaning up TXT record _acme-challenge.%v failed", initiator, *secret.Metadata.Name, *secret.Metadata.Namespace, hostname)
+		// 		log.Info().Err(err).Msgf("[%v] Secret %v.%v - Cleaning up TXT record _acme-challenge.%v failed", initiator, secret.Name, secret.Namespace, hostname)
 		// 	}
 		// }
 
 		// reload secret to avoid object has been modified error
-		err = kubeClient.Get(context.Background(), *secret.Metadata.Namespace, *secret.Metadata.Name, secret)
+		secret, err = kubeClientset.CoreV1().Secrets("").Get(secret.Name, metav1.GetOptions{})
```
Comment (JorritSalverda) [inline main.go:390]: Plug in the `secret.Namespace` inside `.Secrets(...)` to fetch it from the correct namespace.

File: main.go [hunk 3]
```diff
@@ -456,177 +467,170 @@ func makeSecretChanges(kubeClient *k8s.Client, secret *corev1.Secret, initiator
...
-		eventResp.LastTimestamp.Seconds = &secs
-		err = kubeClient.Update(context.Background(), &eventResp)
+		count = eventResp.Count + 1
+		eventResp.Type = eventType
+		eventResp.Action = action
+		eventResp.Reason = reason
+		eventResp.Message = message
+		eventResp.Count = count
+		eventResp.LastTimestamp = metav1.NewTime(now)
+		_, err = kubeClientset.CoreV1().Events(secret.Namespace).Update(eventResp)
```
Comment (JorritSalverda) [inline main.go:568]: Does the controller have event permissions on all namespaces?

File: main.go [hunk 4]
```diff
@@ -456,177 +467,170 @@ func makeSecretChanges(kubeClient *k8s.Client, secret *corev1.Secret, initiator
...
-			Namespace: secret.Metadata.Namespace,
-			CreationTimestamp: &metav1.Time{
-				Seconds: &secs,
-			},
-			Labels: secret.Metadata.Labels,
+	event := &v1.Event{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:              eventName,
+			Namespace:         secret.Namespace,
+			CreationTimestamp: metav1.NewTime(now),
```
Comment (JorritSalverda) [inline main.go:583]: Doesn't this get set automatically?

File: main.go [hunk 5]
```diff
@@ -456,177 +467,170 @@ func makeSecretChanges(kubeClient *k8s.Client, secret *corev1.Secret, initiator
...
-		ReportingComponent: &reportingController,
-		ReportingInstance:  &reportingInstance,
+		EventTime:           metav1.NewMicroTime(now),
+		ReportingController: reportingController,
+		ReportingInstance:   reportingInstance,
 	}
 
-	err = kubeClient.Create(context.Background(), event)
-
+	_, err = kubeClientset.CoreV1().Events(secret.Namespace).Create(event)
```
Comment (JorritSalverda) [inline main.go:605]: Now that the event object has the namespace set I think you can either use `.Events(event.Namespace)` or `.Events("")` to ensure it's slightly more robust and unlikely to go wrong if you every copy this elsewhere.

File: main.go (hunk 3 above)
Comment (vitorarins) [reply main.go:568]: Yes https://github.com/estafette/estafette-letsencrypt-certificate/blob/master/helm/estafette-letsencrypt-certificate/templates/clusterrole.yaml#L29

File: main.go (hunk 4 above)
Comment (vitorarins) [reply main.go:583]: I didn't check that. I saw that usually a [recorder](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/recorder) is used. But I didn't want to make a big change that might break logic.

**Pull request merged**